src/main/java/org/mule/module/apikit/Router.java

```diff
@@ -179,15 +182,15 @@ public ApiSpecification getRaml() {
         });
   }
 
-  private String getRequestPath(HttpRequestAttributes attributes) {
-    // raw request path is encoded only when it's not encoded
-    // if raw request path is decoded always, and encoded again
-    // we can get "not found" error, when raw request path contains %2F (%2F decode-> "/")
-    // example raw request path : "uri-param/AA%2F11%2F00000070" decode-> "uri-param/AA/11/00000070" encode->"uri-param/AA/11/00000070"
-    boolean isEncoded = !attributes.getRequestPath().equals(attributes.getRawRequestPath());
-    String rawRequestPath = isEncoded ? attributes.getRawRequestPath() : UrlUtils.encode(attributes.getRawRequestPath());
-    String path = UrlUtils.getRelativePath(attributes.getListenerPath(), rawRequestPath);
-    return path.isEmpty() ? "/" : path;
+  private String getRequestPath(HttpRequestAttributes attributes) throws BadRequestException {
+    try {
+      String rawRequestPath = URICoder.encodeRequestPath(attributes.getRawRequestPath());
+      String path = UrlUtils.getRelativePath(attributes.getListenerPath(), rawRequestPath);
+      return path.isEmpty() ? "/" : path;
+    } catch (Exception e) {
+      throw new InvalidUriParameterException(e.getMessage());
```

Comment:
Isn't worth to add a simple MUnit test for asserting this scenario?

Reply:
Sure! will be addeed